gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageGrpcReadChannel.java

```diff
@@ -389,6 +382,22 @@ private int readBufferedContentInto(ByteBuffer byteBuffer) {
     return bytesToWrite;
   }
 
+  private boolean canReadFromExistingRequest(ByteBuffer byteBuffer) {
+    if (resIterator == null) {
```

Comment:
does this method need to be synchronised? i.e. can resIterator be set to null from another thread? Same for other instance fields used.